sprout/appliances/tasks.py

```diff
@@ -1627,6 +1628,13 @@ def appliance_yum_update(self, appliance_id):
     appliance.ipapp.update_rhel(reboot=False)
 
 
+@singleton_task()
+def appliance_set_hostname(appliance_id):
+    appliance = Appliance.objects.get(id=appliance_id)
+    if appliance.provider.provider_data.get('type', None) == 'openstack':
+        appliance.ipapp.set_resolvable_hostname()
```

Comment:
REQUIRED: are you sure it will work well w/o callback ?

Reply:
it will now that I wrapped `set_resolvable_hostname` with `logger_wrap` decorator. :)